Adaptors/Memory/src/SessionTable.cs

```diff
@@ -59,28 +60,38 @@ public Task Init(CancellationToken cancellationToken)
     => Task.CompletedTask;
 
   /// <inheritdoc />
-  public Task SetSessionDataAsync(string            rootSessionId,
-                                  Common.Storage.TaskOptions       defaultOptions,
-                                  CancellationToken cancellationToken = default)
+  public Task SetSessionDataAsync(string                     rootSessionId,
+                                  IEnumerable<string>        partitionIds,
+                                  Common.Storage.TaskOptions defaultOptions,
+                                  CancellationToken          cancellationToken = default)
   {
     storage_.TryAdd(rootSessionId,
                     new SessionData(rootSessionId,
                                     SessionStatus.Running,
+                                    partitionIds.ToIList(),
                                     defaultOptions));
     return Task.CompletedTask;
   }
 
   /// <inheritdoc />
-  public Task<bool> IsSessionCancelledAsync(string            sessionId,
-                                            CancellationToken cancellationToken = default)
+  public Task<SessionData> GetSessionAsync(string            sessionId,
+                                           CancellationToken cancellationToken = default)
   {
     if (!storage_.ContainsKey(sessionId))
     {
       throw new SessionNotFoundException($"Key '{sessionId}' not found");
     }
 
-    return Task.FromResult(storage_[sessionId]
-                             .Status == SessionStatus.Canceled);
+    return Task.FromResult(storage_[sessionId]);
+  }
+
+  /// <inheritdoc />
+  public Task<bool> IsSessionCancelledAsync(string            sessionId,
+                                            CancellationToken cancellationToken = default)
+  {
+    return Task.FromResult(this.GetSessionAsync(sessionId,
+                                                cancellationToken)
+                               .Result.Status == SessionStatus.Canceled);
```

Comment:
`.Result` does not wait. The only reason it does work here is because Task is created already completed. This is a safe assumption because the call is to another method of the same class (same file), and even just a few lines above.